lib/publiccloud/azure.pm

```diff
@@ -89,6 +90,10 @@ sub get_storage_account_keys {
     return $key;
 }
 
+sub get_subscription {
```

Comment:
Possibly, but since our credentials are only bound to one single subscription this is not happening.

In the case of having multiple subscriptions it's likely that we are having bigger issues, because then also `terraform` and the other `az` commands will get confused or simply use the default one.

Reply:
OK, we can keep it as it is so when something will change we will at least notice early :D

Reply:
we do have subscription already , please use [$self->provider_client->init();](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure_client.pm#L16) instead of introduction of new function

Reply:
That method relies on having the `PUBLIC_CLOUD_AZURE_SUBSCRIPTION_ID` variable, which I much rather would avoid because it's unnecessary.

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure_client.pm#L34  if this variable undefined you won't be able to continue ANY test anyway , so you MAY rely on the fact that this variable is ALWAYS available

Reply:
No it isn't: https://openqa.suse.de/tests/10221030#settings. And [here](https://openqa.suse.de/tests/10221421#settings) and [here](https://openqa.suse.de/tests/10221033#settings) and [here](https://openqa.suse.de/tests/10221035#settings)

Reply:
Switched to use `$self->provider_client->subscription`, which works, because the `subscription` variable is later on set via the credentials file.